### PR TITLE
🔥 Less aggressive event fire recreation

### PR DESCRIPTION
**Pull request by ericnewcomer**
When adding a contact to a group, we call `handle_update` on that contact which in turn was deleting and recreating all event fires for that contact from scratch. This change makes it so we only delete and (possibly) recreate for the affected group.

### CODE REVIEW

File: temba/contacts/views.py [hunk 1]
```diff
@@ -146,7 +146,7 @@ def clean_query(self):
             raise forms.ValidationError(str(e))
 
     class Meta:
-        fields = "__all__"
+        fields = ("name", "query")
```
Comment (ericnewcomer) [inline temba/contacts/views.py]: Not entirely related, this makes it so we can bring up the edit dialog for large groups

File: temba/contacts/models.py [hunk 2]
```diff
@@ -2285,12 +2285,13 @@ def reevaluate_dynamic_groups(self, for_fields=None, urns=()):
...
         Re-evaluates this contacts membership of dynamic groups. If field is specified then re-evaluation is only
         performed for those groups which reference that field.
+        :returns: the set of groups that were affected
         """
         from .search import evaluate_query
 
         # blocked, stopped or test contacts can't be in dynamic groups
         if self.is_blocked or self.is_stopped or self.is_test:
-            return False
+            return ()
```
Comment (nicpottier) [inline temba/contacts/models.py]: should this be `set()`?

File: temba/contacts/views.py (hunk 1 above)
Comment (nicpottier) [reply temba/contacts/views.py]: Sounds like a useful feature.

File: temba/flows/tests.py [hunk 3]
```diff
@@ -10569,7 +10569,7 @@ def test_flow_batch_start(self):
         stopped.stop(self.admin)
 
         # start our flow, this will take two batches
-        with QueryTracker(assert_query_count=194, stack_count=10, skip_unique_queries=True):
+        with QueryTracker(assert_query_count=184, stack_count=10, skip_unique_queries=True):
```
Comment (rowanseymour) [inline temba/flows/tests.py]: 🍻

**Comment by ericnewcomer** [comment]
Going to do campaign group updates coming in a different PR actually. Ready for re-review please @nicpottier @rowanseymour

**Pull request merged**